cartography/intel/aws/rds.py

```diff
@@ -106,16 +106,96 @@ def load_rds_instances(neo4j_session, data, region, current_aws_account_id, aws_
+    SET sng.name = {DBSubnetGroupName},
+    sng.vpc_id = {VpcId},
+    sng.description = {DBSubnetGroupDescription},
+    sng.status = {DBSubnetGroupStatus},
+    sng.lastupdated = {aws_update_tag}
+    WITH sng
+    MATCH(rds:RDSInstance{id:{DBInstanceArn}})
+    MERGE(rds)-[r:MEMBER_OF_DB_SUBNET_GROUP]->(sng)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag}
+    """
+    if not instance.get('DBInstanceArn'):
+        logger.debug("Expected RDSInstance to have a DBInstanceArn but it doesn't.  Here is the object: %r", instance)
+    elif not instance.get('DBSubnetGroup'):
+        logger.debug("Expected RDSInstance to have a DBSubnetGroup but it doesn't.  Here is the object: %r", instance)
+    else:
+        db_sng = instance.get('DBSubnetGroup')
+        arn = _get_db_subnet_group_arn(region, current_aws_account_id, db_sng.get('DBSubnetGroupName'))
+        neo4j_session.run(
+            attach_rds_to_subnet_group,
+            sng_arn=arn,
+            DBSubnetGroupName=db_sng.get('DBSubnetGroupName'),
+            VpcId=db_sng.get("VpcId", None),
+            DBSubnetGroupDescription=db_sng.get('DBSubnetGroupDescription', None),
+            DBSubnetGroupStatus=db_sng.get('SubnetGroupStatus', None),
+            DBInstanceArn=instance.get('DBInstanceArn'),
+            aws_update_tag=aws_update_tag
+        )
+        _attach_ec2_subnets_to_subnetgroup(neo4j_session, db_sng, region, current_aws_account_id, aws_update_tag)
+
+
+def _attach_ec2_subnets_to_subnetgroup(neo4j_session, db_subnet_group, region, current_aws_account_id, aws_update_tag):
+    """
+    Attach EC2Subnets to the DB Subnet Group.
+
+    From https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_VPC.WorkingWithRDSInstanceinaVPC.html:
+    `Each DB subnet group should have subnets in at least two Availability Zones in a given region. When creating a DB
+    instance in a VPC, you must select a DB subnet group. Amazon RDS uses that DB subnet group and your preferred
+    Availability Zone to select a subnet and an IP address within that subnet to associate with your DB instance.`
+    """
+    attach_subnets_to_sng = """
+    MATCH(sng:DBSubnetGroup{id:{sng_arn}})
+    MERGE(subnet:EC2Subnet{subnetid:{SubnetIdentifier}})
+    ON CREATE SET subnet.firstseen = timestamp()
+    MERGE(sng)-[r:RESOURCE]->(subnet)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag},
+    subnet.availability_zone = {SubnetAvailabilityZone},
+    subnet.lastupdated = {aws_update_tag}
+    """
+    # TODO: Note: RDS data has subnet availability zone data, but I don't think that this is the right place to
+    # TODO: update it.  We should update the availability zone data in the EC2 ingestion part.
+    if not db_subnet_group.get('DBSubnetGroupName'):
+        logger.debug("Expected DBSubnetGroup to have a DBSubnetGroupName but it doesn't.  Here is the object: %r",
+                     db_subnet_group)
```

Comment:
As above, did you see a case where this was missing?

Reply:
Didn't see this particular case, but I also didn't want to make too strong an assumption on the data that would be returned and have this crash a sync.  

There was another case with a different module where we made an assumption and it crashed the sync but I can't recall it right now.